utils/semver/version.go

```diff
@@ -0,0 +1,69 @@
+package semver
+
+import (
+	"context"
+	"embed"
+	"regexp"
+	"strings"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+	"github.com/ARM-software/golang-utils/utils/field"
+	"github.com/ARM-software/golang-utils/utils/filesystem"
+	"github.com/ARM-software/golang-utils/utils/parallelisation"
+)
+
+var (
+	versionRegex = regexp.MustCompile(`Version=(?P<Version>.*)`)
+)
+
+const (
+	DefaultVersionFile = "module.properties"
+)
+
+// DetermineEmbeddedVersion retrieves the version information of a project from embedded files.
+func DetermineEmbeddedVersion(ctx context.Context, fs *embed.FS, versionFilePath *string) (version string, err error) {
+	eFs, err := filesystem.NewEmbedFileSystem(fs)
+	if err != nil {
+		return
+	}
+	version, err = DetermineVersionFromFile(ctx, eFs, field.OptionalString(versionFilePath, DefaultVersionFile))
+	return
+}
+
+// DetermineVersionFromFile retrieves the information about the version of a project based on a version file generated by [autoversion](https://github.com/ARMmbed/autoversion).
+// The Canonical version is returned.
+func DetermineVersionFromFile(ctx context.Context, fs filesystem.FS, versionFilePath string) (version string, err error) {
+	err = parallelisation.DetermineContextError(ctx)
+	if err != nil {
+		return
+	}
+	if fs == nil {
+		err = commonerrors.UndefinedVariable("file system", "")
+		return
+	}
+	if strings.TrimSpace(versionFilePath) == "" {
```

Comment:
```suggestion
	versionFilePath := strings.TrimSpace(versionFilePath)
	if versionFilePath == "" {
```

It might better to strip it here since then it isn't stripped when you do FS.Exists later https://github.com/ARM-software/golang-utils/blob/f9cd35c9f9f09e9abb1fd3a5e7db6b59673c4105/utils/filesystem/files.go#L584